src/dict_merge.py

```diff
@@ -0,0 +1,176 @@
+    geo_dict = {'zip':'zipcodes', 'county':'counties'}
+    for k,v in d_dict.items():
+        geo_str = geo_dict[k]
+        fp =  'shape_files/ILgeojson.json' if geo_str != 'counties' else 'tests/resources/ILgeojson_county.json'
+        #only add bins to geojson properties
+        geo_json, g_map = getGeoJson(fp=fp, param=geo_str)
+        #breakpoint()
+        merged_json = merge(v)
+        final_json = {}
+        final_json[geo_str] = mergeGeoJson(geo_json, g_map, merged_json)
+        final_json_ls.append(final_json)
+        # breakpoint()
+        with open(F'final_jsons/merged{k}_output.json','w') as f:
+            json.dump(final_json, f, separators=(',',':'))
+    
+    #saves merged_dict to file
+    with open(F'final_jsons/merged_output.json','w') as f:
+        #geojson = getGeoJson()
+        merged_dict = {k:v for d in final_json_ls for k, v in d.items()}
+        # breakpoint()
+        json.dump(merged_dict, f, separators=(',',':'))
+
+    return merged_dict
+
+
+def merge(dictList = list()):
+    '''
+    Merges data dictionaries
+    input: list of geo data dictionaries (list)
+            dictionary format {geo_area:{'geo_code':{'metric1':1,...},...}}
+            within each dictionary assumes all geo-areas include the same metrics
+    output: merged geo data dictionary, same format
+    '''
+    #TODO What happens if some geo areas do not have all the data elements?
+    #Error handled: prints geo area not in list index
+    #Do we want to add the missing data with null values?
+    # breakpoint()
+    mergedDict = dict()
+    
+    #Get all the geocodes in both datasets
+    mergedKeys = set()
+    for d in dictList:
+        # breakpoint()
+        mergedKeys.update(set(d.keys()))
+    
+    #loop through all geocodes
+    for k in mergedKeys:
+        #creates geocode key for mergedDict
+        mergedDict[k] = dict()
+        for i, d in enumerate(dictList):
+            # breakpoint()
+            #if the geocode is not in the dictionary, print to console and continue
+            #TODO add missing geocode metric here?
+            try:
+                value = list(d[k])
+            except:
+                print(k, F' not in dictList[{i}]')
+                continue
+            for v in value:
+                #if the metric is already in the dictionary it is a duplicate
+                if v in mergedDict[k]:
+                    #if the duplicate key includes an identical value continue
+                    if mergedDict[k][v] == d[k][v]:
+                        continue
+                    #Otherwise, raise exception and return metric name
+                    e = 'Duplicate metric key: rename to avoid overwrite'
+                    raise Exception(e, v)
+                #The metric is not in the dictionary, add it
+                else:
+                    mergedDict[k][v] = d[k][v]
+    return mergedDict
+
+def getGeoJson(fp = 'shape_files/ILgeojson.json', param=""):
```

Comment:
We should change these function names to snake_case. get_geo_json. A linter would enforce this.